exercises/danny_hindson/ex02_oo_basics/src/Shape.h

```diff
@@ -0,0 +1,27 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#ifndef SHAPE_H_
+#define SHAPE_H_
+
+#include <string>
+
+class Shape {
+public:
+  Shape(std::string ShapeType, int nsides) : ShapeType(ShapeType), nsides(nsides) {};
+  virtual ~Shape() {};
```

Comment:
Again this is perfectly valid but C++11 has introduced the `default` qualifier to avoid having to rewrite what the compiler would generate when we are forced to include things like destructors in this manner:
```c++
virtual ~Shape() = default;
```